pkg/process/procutil/process_linux.go

```diff
@@ -175,6 +175,38 @@ func (p *Probe) ProcessesByPID(now time.Time) (map[int32]*Process, error) {
 	return procsByPID, nil
 }
 
+// StatsWithPermByPID returns the stats that require extra permission to collect for each process
+// the parameter `returnZeroVals` controls whether to include all zero stats or not in the return map,
+// not returning all zero stats would help process-agent to deserialize less data
+func (p *Probe) StatsWithPermByPID(returnZeroVals bool) (map[int32]*StatsWithPerm, error) {
```

Comment:
How do you differentiate between stats for processes that we failed to collect versus stats that are just zero?

Reply:
This is done in this PR: https://github.com/DataDog/datadog-agent/pull/7511